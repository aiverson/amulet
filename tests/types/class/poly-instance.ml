class foo 'a begin end
class trivial begin end

instance foo (trivial => 'a) begin end
