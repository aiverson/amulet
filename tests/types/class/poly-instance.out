poly-instance.ml[4:1 ..4:38]: error (E2006)
  │ 
4 │ instance foo (trivial => 'a) begin end
  │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Illegal use of polymorphic type trivial => 'a
    as argument to the type constructor Foo
poly-instance.ml[4:1 ..4:38]: Use amc explain 2006 for more information
